### PR TITLE
feat: fix error message to be more explicit

### PR DESCRIPTION
**Pull request by eddyxu**
Closes https://github.com/lancedb/lancedb/issues/502

### CODE REVIEW

File: rust/lance/src/io/exec/knn.rs [hunk 1]
```diff
@@ -171,7 +171,7 @@ impl KNNFlatExec {
         if !is_vector {
             return Err(Error::IO {
                 message: format!(
-                    "KNNFlatExec node: query column {} is not a vector",
+                    "KNNFlatExec node: query column {} is not a vector (float32 fixed size array)",
                     query.column
```
Comment (wjones127) [inline rust/lance/src/io/exec/knn.rs:175]: Maybe we could also name the data type we got?

```suggestion
                    "KNNFlatExec node: query column {} is not a vector. Expected float32 fixed-size array, got {}",
                    query.column, field.data_type()
```

**Review by westonpace — COMMENTED** [review]
Can we print what type it is?  Something like "KNNFlatExec node: query column has type XYZ which is not a vector (float32 fixed size array)"

Also, I'm a little surprised we get this far.  A user is probably going to wonder what "KNNFlatExec node" means.  It's not a deal breaker (they can just ignore that) but we might want to validate the query type earlier.

**Comment by eddyxu** [comment]
> Also, I'm a little surprised we get this far. A user is probably going to wonder what "KNNFlatExec node" means. It's not a deal breaker (they can just ignore that) but we might want to validate the query type earlier.

make sense. Lemme take a look

File: rust/lance-arrow/src/floats.rs [hunk 2]
```diff
@@ -40,7 +40,7 @@ impl FloatArray for BFloat16Array {
 impl<T: ArrowPrimitiveType> FloatArray for PrimitiveArray<T>
 where
     T::Native: Float + FromPrimitive,
-    PrimitiveArray<T>: From<Vec<T::Native>>,
+    Self: From<Vec<T::Native>>,
```
Comment (eddyxu) [inline rust/lance-arrow/src/floats.rs:43]: Clippy error

**Review by westonpace — APPROVED** [review]

**Pull request merged**